src/server/JasmineGraphInstanceService.cpp

```diff
@@ -3584,3 +3650,309 @@ map<long, map<long, unordered_set<long>>> calculateEgoNet(string graphID, string
+
+                double existingChildRank = 0;
+                if (rankMapItr != rankMap.end()) {
+                    existingChildRank = rankMapItr->second;
+                    double finalRank = existingChildRank + distributedRank;
+
+                    rankMapItr->second = finalRank;
+                } else {
+                    double finalRank = existingChildRank + distributedRank;
+                    rankMap.insert(std::make_pair(itr, finalRank));
+
+                }
+            }
+        }
+
+        count++;
+    }
+
+    map<double, long> rankMapResults;
+
+    if (k == -1) {
+        instance_logger.log("Page rank is not implemented", "info");
+    } else {
+        std::string resultTree = "";
+        instance_logger.log("tree size " + to_string(rankMap.size()), "info");
+
+        for (map<long, double>::iterator rankMapItr = rankMap.begin(); rankMapItr != rankMap.end(); ++rankMapItr) {
+            rankMapResults.insert(std::make_pair(rankMapItr->second, rankMapItr->first));
+            resultTree.append(to_string(rankMapItr->first) + ":" + to_string(rankMapItr->second) + ",");
+        }
+
+        int count = 0;
+        for (map<double, long>::iterator rankMapItr = rankMapResults.end();
+             rankMapItr != rankMapResults.begin(); --rankMapItr) {
+
+            if (count > 50) {
+                break;
+            }
+            instance_logger.log(to_string(rankMapItr->first) + ":" + to_string(rankMapItr->second) + ",", "info");
+            count++;
+        }
+
+    }
+
+}
+
+map<long, float> getAuthorityScoresWorldToLocal(string graphID, string partitionID, int serverPort,
+                                                string graphVertexCount, JasmineGraphHashMapLocalStore localDB,
+                                                JasmineGraphHashMapCentralStore centralDB,
+                                                map<long, unordered_set<long>> graphVertexMap,
+                                                std::vector<string> workerSockets, long worldOnlyVertexCount) {
+
+    map<long, unordered_set<long>> worldToLocalVertexMap;
+    map<long, long> fromDegreeDist;
+
+    for (vector<string>::iterator workerIt = workerSockets.begin(); workerIt != workerSockets.end(); ++workerIt) {
+        std::vector<string> workerSocketPair;
+        stringstream wl(*workerIt);
+        string intermediate;
+        while (getline(wl, intermediate, ':')) {
+            workerSocketPair.push_back(intermediate);
+        }
+
+        Utils utils;
+        std::string aggregatorFilePath = utils.getJasmineGraphProperty("org.jasminegraph.server.instance.datafolder");
+        std::string centralGraphIdentifier = graphID + +"_centralstore_" + workerSocketPair[2];
+
+        std::string centralStoreFile = aggregatorFilePath + "/" + centralGraphIdentifier;
+        instance_logger.log("###INSTANCE### centralstore " + centralStoreFile, "info");
+
+        struct stat s;
```

Comment:
Can we have a better variable name than `s`?

Reply:
fixed

Reply:
Ok